akka-actor/src/main/scala/akka/io/Dns.scala

```diff
@@ -89,3 +89,11 @@ class DnsExt(system: ExtendedActorSystem) extends IO.Extension {
 
   def getResolver: ActorRef = manager
 }
+
+object IpVersionSelector {
+  def getInetAddress(ipv4: Option[Inet4Address], ipv6: Option[Inet6Address]): Option[InetAddress] =
+    sys.props.get("java.net.preferIPv6Addresses") match {
+      case Some(value: String) if ("true".equals(value)) ⇒ ipv6 orElse ipv4
+      case _ ⇒ ipv4 orElse ipv6
+    }
```

Comment:
Good point, I've updated the PR with the proposed change. Thanks for taking care of the tests.